MissionMonitor/ConfigurationWindow.xaml.cs

```diff
@@ -50,6 +53,77 @@ private void warningChanged(object sender, TextChangedEventArgs e)
             }
         }
 
+        private void getRoute(object sender, RoutedEventArgs e)
+        {
+            Button updateButton = (Button)sender;
+            updateButton.Foreground = Brushes.Red;
+            updateButton.FontWeight = FontWeights.Bold;
+
+            Thread getRouteThread = new Thread(() =>
+            {
+                string nextSystem = missionMonitor().GetMissionsRoute();
+                Dispatcher?.Invoke(() =>
+                {
+                    updateButton.Foreground = Brushes.Black;
+                    updateButton.FontWeight = FontWeights.Regular;
+
+                    // If 'next system' found, send to clipboard
+                    if (nextSystem != null)
+                    {
+                        Clipboard.SetData(DataFormats.Text, nextSystem);
+                    }
+                });
+            })
+            {
+                IsBackground = true
+            };
+            getRouteThread.Start();
+        }
+
+        private void nextInRoute(object sender, RoutedEventArgs e)
+        {
+            string nextSystem = missionMonitor().SetNextRoute();
+
+            // If 'next system' found, send to clipboard
+            if (nextSystem != null)
+            {
+                Clipboard.SetData(DataFormats.Text, nextSystem);
+            }
+
+        }
+
+        private void updateRoute(object sender, RoutedEventArgs e)
+        {
+            Button updateButton = (Button)sender;
+            updateButton.Foreground = Brushes.Red;
+            updateButton.FontWeight = FontWeights.Bold;
+
+            Thread updateRouteThread = new Thread(() =>
+            {
+                string nextSystem = missionMonitor().UpdateRoute();
+                Dispatcher?.Invoke(() =>
+                {
+                    updateButton.Foreground = Brushes.Black;
+                    updateButton.FontWeight = FontWeights.Regular;
+
+                    // If 'next system' found, send to clipboard
```

Comment:
Should setting the clipboard be optional? If not, do we say in the UI that we'll be altering the clipboard?